geoutils/georaster/raster.py

```diff
@@ -624,11 +626,7 @@ def _overloading_check(
         if (nodata1 is not None) and (out_dtype == dtype1):
             out_nodata = nodata1
 
-        # Convert output data to correct dtype and masked_array
-        if isinstance(other_data, np.ndarray):
-            other_data = np.ma.asarray(other_data).astype(out_dtype, copy=False)
-
-        self_data = self.data.astype(out_dtype)
+        self_data = self.data
```

Comment:
Are you sure the call to `astype` is not needed anymore? I guess if the tests were well designed (hum hum) they would fail otherwise...

Reply:
Yes, NumPy arithmetic already takes care of it all, we just have to pass it.
The tests on arithmetic operations are actually pretty thorough! (especially with the new `__eq__`!)